src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -220,9 +223,18 @@ public void onRequestFailure(final SearchPhaseContext context, final SearchReque
         constructSearchQueryRecord(context, searchRequestContext);
     }
 
-    private void constructSearchQueryRecord(final SearchPhaseContext context, final SearchRequestContext searchRequestContext) {
+    private boolean skipSearchRequest(final SearchRequestContext searchRequestContext) {
         // Skip profile queries
         if (searchRequestContext.getRequest().source().profile()) {
+            return true;
+        }
+
+        Set<String> excludedIndices = new HashSet<>(clusterService.getClusterSettings().get(TOP_N_QUERIES_EXCLUDED_INDICES));
```

Comment:
Store HashSet in a class variable so it does not need to be reinitialized on every request